File: loglib/src/main/java/com/robog/loglib/database/LogBeanDao.java
package com.robog.loglib.database;

import com.robog.loglib.LogBean;

import java.util.List;

/**
 * Created by yuxingdong on 2018/6/21.
 */
public interface LogBeanDao {

    void save(List<LogBean> logBeans);

    /**
     * 降序
     * @return
     */
    List<LogBean> getAll();

    boolean checkThreshold();

    void deleteAll();
}

File: loglib/src/main/java/com/robog/loglib/database/LogBeanDao.kt
package com.robog.loglib.database

import com.robog.loglib.LogBean

/**
 * Created by yuxingdong on 2018/6/29.
 */
interface LogBeanDao {

    fun getAll(): List<LogBean>

    fun save(logBeans: List<LogBean>)

    fun checkThreshold(): Boolean

    fun deleteAll()
}
